contributors/design-proposals/live-and-inplace-vertical-scaling.md

```diff
@@ -0,0 +1,325 @@
+    }
+}
+```
+
+## Workflow
+
+This describes the sequence of a pod-level vertical scaling example to resize the CPU resource of a pod from 1 to 2.
+
+![Process](live-and-inplace-vertical-scaling.png)
+
+0. A pod has 1 CPU.
+
+1. A client requests resource resizing on the pod with a new PodSpec with 2 CPU.
+
+2. The API server updates the PodSpec with ResizeRequest on etcd
+
+3. The PodSpec is updated on etcd.
+
+4. The Scheduler checks if the resizing is feasible and, if so, issues a Resizing API operation with the ResizeRequest of the “Accepted” status.
+
+5. The API server updates the PodSpec with the new resource requirement on etcd and also modifies the ResizeRequest of the PodSpec to “Accepted”.
+
+6., 7. The Kubelet updates the PodResized condition to “Accepted”.
+
+8. The Kubelet detects the change of resource requirements on the container and updates the cgroup configuration of the container via UpdateContainerResources CRI interface.
+
+9. The Kubelet modifies the status of the PodResized condition to Done after every update on the cgroup configuration of all containers to resize is done. 
+
+10. It completes to resize the pod to have 2 CPUs.
+
+
+## Implementation Phases
+
+* Phase 1 - Introduce live and in-place vertical scaling on a pod
+
+Status: In progress, a working prototype implemented originally in the master branch (v1.10-alpha)
+
+The working code is available at [https://github.com/YoungjaeLee/kubernetes](https://github.com/YoungjaeLee/kubernetes) (the qos-master branch)
+
+* Phase 2 - Adding support for live and in-place vertical scaling in StatefulSet
+
+Status: In progress in the master branch (v1.10-alpha)
+
+
+# Related issues
+
+1. QoS class change by resize is not supported.
+
+For each of the Burstable and the Best-effort class, the Kubelet maintains a class-level cgroup under the ‘kubepods’ cgroup, which is the parent cgroup of pods of each QoS class. (e.g. kubepods/burstable is the parent cgroup of Burstable pods).
+So, in order to change the QoS class of a pod, it needs not only to resize resources, but also to change its parent cgroup properly.
+But, once a pod is created, its parent cgroup cannot be changed with the current Docker API. So, the QoS-class of a pod cannot be changed by resource resizing.
+
+2. Memory-resizing to change a request value might not take effect for Burstable pods.
+
+For Burstable pods, a request value for memory resource determines the value of a score for the OOM killer, but Docker doesn’t support to change dynamically the score of an existing container.
+So, the change to a memory request value doesn’t take effect for Burstable pods on the OOM killer’s behavior.
+But, for Guaranteed and Best-effort pods, this is not an issue because the score is fixed regardless of its memory request value. (in this case, the memory request value is used only for admission control by Scheduler) 
+
+3. Memory-resizing to decrease its limit may fail on the Kubelet in some circumstances.
```

Comment:
We have a control loop that you can enable at the QoS tier to keep trying to induce reclaim.  I worry that these issues unless really well understood will cause confusion for users and operators.